2019-Q4-PyPI/RFI.md

```diff
@@ -0,0 +1,164 @@
+# PyPI Quarter 4 2019 Request for Information
+
+The [Python Software Foundation](https://python.org/psf-landing) [Packaging Working Group](https://www.python.org/psf/committees/#packaging-work-group) has [received a grant](https://pyfound.blogspot.com/2018/12/upcoming-pypi-improvements-for-2019.html) from [Facebook Research](https://research.fb.com) to implement advanced security features for PyPI.
+
+[PyPI](https://pypi.org) is a foundational component of the [Python](https://python.org) ecosystem and broader computer software and technology landscape. This project aims to improve the security of PyPI for all users worldwide, whether they are direct users like project maintainers and [`pip install`](https://pip.pypa.io/en/stable/)ers or indirect users. The impact of this work will be to implement long-desired security features for the service (see below).
+
+We plan to begin the project in Quarter 4 of 2019. Because of the size of the project, funding has been allocated to secure one or more contractors to complete the development, testing, verification, and assist in the rollout of necessary features.
+
+Please read this RFI and respond to let us know if you have questions, or may be interested in applying to work on the project.
+
+## Timeline
+
+|Date|Milestone|
+|---|---|
+|TBD|Request for Information period opens.|
+|TBD|Request for Information period closes.|
+|TBD|Request for Proposal period opens. (? weeks after RFI close)|
+|TBD|Request for Proposal period closes. (? days after RFP open)|
+|TBD|Date proposals will have recieved a decision. (? weeks after RFP close)|
+|TBD|Contracts for accepted proposals should be finalized.|
+|TBD|Contract work commences.|
```

Comment:
Here's my suggestion:

|August 19|Request for Information period opens.|
|September 5|Request for Information period closes.|
|September 16|Request for Proposal period opens. (11 days after RFI close)|
|October 18|Request for Proposal period closes. (basically 1 month after RFP open)|
|November 1|Date proposals will have received a decision. (2 weeks after RFP close)|
|November 11|Contracts for accepted proposals should be finalized.|
|December 1|Contract work commences.|

This allows us to start in Q4. It gives us 11 days to talk with the Packaging WG, Facebook, and any other stakeholders about what we learn from the RFI, to revise the RFP. And it gives us room to extend the RFP deadline by 1 week, to Oct 25, and still give us a week to review proposals before making a decision by Nov. 1.

Reply:
I missed that we may need more than a month to finalize contracts (especially around US Thanksgiving) -- and I think we can shorten the RFP window to 3 weeks:

|August 19|Request for Information period opens.|
|September 5|Request for Information period closes.|
|September 16|Request for Proposal period opens. (11 days after RFI close)|
|October 8|Request for Proposal period closes. (3 weeks after RFP open)|
|October 22|Date proposals will have received a decision. (2 weeks after RFP close)|
|November 31|Contracts for accepted proposals should be finalized.|
|December 1|Contract work commences.|